core/src/main/java/org/elasticsearch/index/query/functionscore/FunctionScoreQueryBuilder.java

```diff
@@ -450,8 +459,34 @@ public FilterFunctionBuilder rewrite(QueryRewriteContext context) throws IOExcep
         return this;
     }
 
-    public static FunctionScoreQueryBuilder fromXContent(Function<String, ScoreFunctionParser<?>> scoreFunctionLookup,
-            QueryParseContext parseContext) throws IOException {
+    /**
+     * Looks up {@linkplain ScoreFunctionParser}s based on name and context. Needs context to check the ParseFieldMatcher and report nice
+     * errors.
+     */
+    public static class ScoreFunctionsRegistry {
```

Comment:
I might have broken it in the last PR about this. But, anyway, I think I fixed it in my last commit. Now the registry is exposed.